docs/docs/forms.mdx

```diff
@@ -297,55 +297,37 @@ actions:
 - validate_restaurant_form
 ```
 
-When the form is executed it will run your custom action. In your custom action
-you can either
+When the form is executed it will run your custom action.
 
-- validate already extracted slots. You can retrieve them from the tracker by running
-  `tracker.form_slots_to_validate`.
-- use [Custom Slot Mappings](./forms.mdx#slot-mappings) to extract slot values .
-
-After validating the extracted slots, return `SlotSet` events for them. If you want
-to mark a slot as invalid return a `SlotSet` event which sets the value to `None`.
-Note that if you don't return a `SlotSet` for an extracted slot, Rasa Open Source
-will assume that the value is valid.
+This custom action can extend `FormValidationAction` class to simplify
+the process of validating extracted slots. In this case, you need to write functions
+named `validate_{slot_name}` for every extracted slot.
 
 The following example shows the implementation of a custom action
-which validates that every extracted slot is valid.
+which validates that the slot named `slot1` is valid.
 
 ```python
-from typing import Dict, Text, List, Any
+from typing import Text, List, Any
 
 from rasa_sdk import Tracker
-from rasa_sdk.events import EventType
-from rasa_sdk.executor import CollectingDispatcher
-from rasa_sdk import Action
 from rasa_sdk.events import SlotSet
+from rasa_sdk.executor import CollectingDispatcher
+from rasa_sdk.forms import FormValidationAction
+from rasa_sdk.types import DomainDict
 
 
-class ValidateRestaurantForm(Action):
+class ValidateRestaurantForm(FormValidationAction):
     def name(self) -> Text:
         return "validate_restaurant_form"
 
-    def run(
-        self, dispatcher: CollectingDispatcher, tracker: Tracker, domain: Dict
+    def validate_slot1(
+        self,
+        slot_value: Any,
+        dispatcher: CollectingDispatcher,
+        tracker: Tracker,
+        domain: DomainDict,
     ) -> List[EventType]:
-        extracted_slots: Dict[Text, Any] = tracker.form_slots_to_validate()
-
-        validation_events = []
-
-        for slot_name, slot_value in extracted_slots:
-            # Check if slot is valid.
-            if is_valid(slot_value):
-                validation_events.append(SlotSet(slot_name, slot_value))
-            else:
-                # Return a `SlotSet` event with value `None` to indicate that this
-                # slot still needs to be filled.
-                validation_events.append(SlotSet(slot_name, None))
-
-        return validation_events
-
-    def is_valid(slot_value: Any) -> bool:
-        # Implementation of the validate function.
+        return [SlotSet("slot1", "value")]
```

Comment:
Can we please give an example of a real validation here so that users see what to do in case the slot is valid and what to in case the slot is invalid?

Reply:
just realized that this is different to your implementation (you return a dict there).